api/src/main/java/com/intern/api/client/DataBaseClient.java

```diff
@@ -17,17 +31,47 @@
   final MongoCollection<Document> collection;
 
   public Document deleteMessage(Long messageId) {
-    Document doc = collection.find(eq("message_id", messageId)).first();
-    collection.deleteOne(doc);
+    Document doc = collection.findOneAndDelete(eq("message_id", messageId));
     log.debug("Document: {} delete successfully", doc.toJson());
     return doc;
   }
 
-  public ArrayList<Document> getMessages() {
+  public List<Document> getMessages() {
     return collection.find().into(new ArrayList<>());
   }
 
-  public ArrayList<Document> getMessagesInChat(Long chatId) {
+  public List<Document> getMessagesByChatId(Long chatId) {
     return collection.find(eq("chat.id", chatId)).into(new ArrayList<>());
   }
+
+  public List<Document> getPreviousMessages(Long userId) {
+    return collection.find(eq("from.id", userId)).limit(10).into(new ArrayList<>());
```

Comment:
Does this operation loads the whole history of user's messages and then filters them? Can we limit on query-level?

Reply:
Also make limit number as parameter. Use default value on controller level.